scripts/services/check-removed-services.js

```diff
@@ -0,0 +1,133 @@
+ * if the JSON file is successfully read and parsed. Returns `null` if there's an error during the process.
+ */
+const getJsonObjects = async (jsonFilePath) => {
+    try {
+        const jsonFileContent = await fs.readFile(jsonFilePath);
+        const jsonObjects = JSON.parse(jsonFileContent);
+        return jsonObjects.blocked_services;
+    } catch (error) {
+        console.error('Error while reading JSON file:', error);
+        return null;
+    }
+};
+
+/**
+ * Gets the names of services from JSON file after normalization.
+ *
+ * @param {Promise<Array<object>>} jsonData - An array of objects from services.json file.
+ * @returns {Promise<Array<string>>} An array of normalized service id form JSON.
+ */
+const getJsonObjectNames = async (jsonData) => {
+    // get array with id's from json data
+    const jsonServicesId = jsonData.map((service) => service.id);
+    // format file names
+    const formattedServicesId = jsonServicesId.map(normalizeFileName);
+    // return sorted array
+    return formattedServicesId.sort();
+};
+
+/**
+ * Gets the names of YML file from the services folder.
+ *
+ * @param {string} servicesFolderPath - The path to the folder containing YML service files.
+ * @returns {Promise<Array<string>>} An array of normalized yml file names.
+ */
+const getYmlFileNames = async (servicesFolderPath) => {
+    // get all dir names from services folder
+    const ymlFiles = await fs.readdir(servicesFolderPath);
+    // get the file names without its extension
+    const ymlFileNames = ymlFiles.map((ymlFile) => path.parse(ymlFile).name);
+    // format file names
+    const formattedServiceNames = ymlFileNames.map(normalizeFileName);
+    // return sorted array
+    return formattedServiceNames.sort();
+};
+
+/**
+ * Write removed services objects into YML files.
+ *
+ * @param {Array<object>} removedObjects - Array of objects that should be written in separate YML files.
+ */
+const writeRemovedServices = async (removedObjects) => {
+    if (removedObjects.length === 0) {
+        return;
+    }
+    const [removedObject, ...restObjects] = removedObjects;
+    await fs.writeFile(
+        path.join(`${servicesDir}/${removedObject.id}${YML_FILE_EXTENSION}`),
+        yaml.dump(removedObject, { lineWidth: -1 }),
+    );
+    if (removedObjects.length > 1) {
+        await writeRemovedServices(restObjects);
+    }
+};
+
+/**
+ * Rewrites YAML files for removed services.
+ *
+ * @param {Array<string>} removedServicesNames - An array of removed normalized service names.
+ * @param {Array<object>} jsonServicesData - An array of json data objects
+ * @throws {Error} If there is an error while rewriting file.
+ */
+const rewriteYMLFile = async (removedServicesNames, jsonServicesData) => {
+    try {
+        // Get services objects from the json data, that have been deleted in services folder.
+        const removedServicesObjects = removedServicesNames
```

Comment:
In this case, you normalize JSON services too many times unnecessarily. Please consider using something similar:

```js
// No need to do anything if there are no removed services are specified or if there is no JSON services data.
// FIXME: Can they be empty? If so, you can avoid any unnecessary normalization
if (removedServicesNames.length === 0 || jsonServicesData.length === 0) {
    return;
}

// Create a set of normalized service names from the JSON services data.
const normalizedJsonServiceNames = new Set(jsonServicesData.map(({ id }) => normalizeFileName(id)));

// Filter the removed services names to get only those that are present in the JSON services data.
const removedServicesObjects = removedServicesNames.filter((name) => normalizedJsonServiceNames.has(name));
```

Also, I think it would be better to use `removedServiceNames` here instead of `removedServicesNames`

Reply:
`rewriteYMLFile` is called when `removedServicesNames > 0`

````
if (differences.length > 0) {
        await rewriteYMLFile(differences, jsonDataObjects);
        console.log(`These services have been removed: ${differences.join(', ')}, and were rewritten`);
    }
````
`jsonServicesData` cannot be 0, then we would have nothing to take `removedServicesNames` from

Reply:
Ok, then the check is not necessary

Reply:
this script extract an object from `services.json`, which by normalized `id` is equal to the normalized `yml` filename.  We need to keep the `id`, because the `id` should not be changed. For example `yml` file name `epic-games.yml`, inside `id: epic_games` and in services.json `id: epic_games`. So we bring the filename and id to the same form and match, then take the fully untouched object from `services.json`.
Slava suggested in the future to get rid of `id` inside the `yml` file and take `id` directly from the `yml` filename, I will add #TODO comment